runtime/integration-tests/pendulum/src/tests.rs

```diff
@@ -0,0 +1,350 @@
+			&ALICE.into(),
+		);
+	});
+
+	Relay::execute_with(|| {
+		assert_ok!(polkadot_runtime::XcmPallet::reserve_transfer_assets(
+			polkadot_runtime::RuntimeOrigin::signed(ALICE.into()),
+			Box::new(Parachain(2094).into().into()),
+			Box::new(Junction::AccountId32 { network: NetworkId::Any, id: ALICE }.into().into()),
+			Box::new((Here, transfer_amount).into()),
+			0
+		));
+	});
+
+	PendulumParachain::execute_with(|| {
+		use pendulum_runtime::{RuntimeEvent, System};
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Tokens(orml_tokens::Event::Deposited { .. })
+		)));
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::DmpQueue(cumulus_pallet_dmp_queue::Event::ExecutedDownward { .. })
+		)));
+	});
+
+	PendulumParachain::execute_with(|| {
+		assert_eq!(
+			pendulum_runtime::Tokens::balance(
+				pendulum_runtime::PendulumCurrencyId::XCM(0),
+				&ALICE.into()
+			),
+			orml_tokens_before + transfer_amount - DOT_FEE
+		);
+	});
+}
+
+#[test]
+fn transfer_polkadot_from_pendulum_to_relay_chain() {
+	MockNet::reset();
+
+	let transfer_dot_amount: Balance = dot(10);
+
+	Relay::execute_with(|| {
+		let after_bob_free_balance = polkadot_runtime::Balances::free_balance(&BOB.into());
+		assert_eq!(after_bob_free_balance, dot(100));
+	});
+
+	PendulumParachain::execute_with(|| {
+		assert_ok!(pendulum_runtime::XTokens::transfer(
+			pendulum_runtime::RuntimeOrigin::signed(BOB.into()),
+			pendulum_runtime::PendulumCurrencyId::XCM(0),
+			transfer_dot_amount,
+			Box::new(
+				MultiLocation::new(
+					1,
+					Junctions::X1(Junction::AccountId32 { id: BOB, network: NetworkId::Any })
+				)
+				.into()
+			),
+			WeightLimit::Limited(4_000_000_000),
+		));
+	});
+
+	PendulumParachain::execute_with(|| {
+		use pendulum_runtime::{RuntimeEvent, System};
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Tokens(orml_tokens::Event::Withdrawn { .. })
+		)));
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::XTokens(orml_xtokens::Event::TransferredMultiAssets { .. })
+		)));
+	});
+
+	Relay::execute_with(|| {
+		use polkadot_runtime::{RuntimeEvent, System};
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Balances(pallet_balances::Event::Withdraw { .. })
+		)));
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Balances(pallet_balances::Event::Deposit { .. })
+		)));
+
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::Ump(polkadot_runtime_parachains::ump::Event::ExecutedUpward { .. })
+		)));
+	});
+
+	Relay::execute_with(|| {
+		let after_bob_free_balance = polkadot_runtime::Balances::free_balance(&BOB.into());
+		assert_eq!(after_bob_free_balance, dot(100) + transfer_dot_amount - FEE);
+	});
+}
+
+#[test]
+fn statemint_transfer_incorrect_asset_to_pendulum_fails() {
```

Comment:
It's a bit hard to understand what this test is about. Why is the asset `XCM(1)` 'incorrect' and what would we expect? Maybe add some comments to the code.

Reply:
Nice, thanks for the great comment. 
Should we move it above the declaration of the test function ie line 135 maybe?

Reply:
Done.